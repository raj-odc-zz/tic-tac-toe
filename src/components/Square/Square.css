.Field {
  width: 33%;
  height: 33%;
  float: left;
  transform: scale(.95, .95);
  transition: background 0.2s ease;
  background: #333;
}

.Field:hover {
  animation: pulsate infinite .8s ease-in-out;
}

.Field.is-player-1 {
  transform: scale(.9, .9);
  background: #64CEAA;
}

.Field.is-player-2 {
  transform: scale(.9, .9);
  background: #FD6C6C;
}
