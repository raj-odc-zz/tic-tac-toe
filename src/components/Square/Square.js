/*
 * Replace the `alert` with the actual dispatch later on
 *
 * You already have utilit classes available to mark the current square selected by a player:
 *  .is-player-0
 *  .is-player-1
 *
 * Check out the src/index.html for pre-defined basic css classes.
 */
import React from 'react';

import './Square.css';

const Square = (props) => {
  const { squareValue, onClick, squareIndex } = props;
  let squareClassName = 'Field';
  if (squareValue) {
    squareClassName += ` is-player-${squareValue}`;
  }
  return (
    <div
      className={squareClassName}
      role="presentation"
      onClick={() => onClick(squareIndex)}
      onKeyDown={() => {}}
    />
  );
};

export default Square;
