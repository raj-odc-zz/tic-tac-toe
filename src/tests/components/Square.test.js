import * as React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Square from '../../components/Square/Square.js';

describe('<Square />', () => {
  let wrapper;

  it('should load the default square',() => {
    wrapper = mount(<Square squareValue={null} onClick={() => {}} squareIndex={1} />);
    expect(wrapper.find('.Field').length).toEqual(1)
  })

  it('should update with class is-player-1',() => {
    wrapper = mount(<Square squareValue={1} onClick={() => {}} squareIndex={1} />);
    expect(wrapper.find('.is-player-1').length).toEqual(1)
  })

  it('should update with class is-player-2',() => {
    wrapper = mount(<Square squareValue={2} onClick={() => {}} squareIndex={1} />);
    expect(wrapper.find('.is-player-2').length).toEqual(1)
  })

  it('renders correctly', () => {
    const tree = renderer
      .create(<Square squareValue={2} onClick={() => {}} squareIndex={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});