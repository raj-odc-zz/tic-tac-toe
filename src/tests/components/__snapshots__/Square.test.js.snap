// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> renders correctly 1`] = `
<div
  className="Field is-player-2"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
/>
`;
